pkg/resource/deploy/plan_executor.go

```diff
@@ -246,15 +271,15 @@ func (pe *planExecutor) performDeletes(ctx context.Context, opts Options) result
 
 // handleSingleEvent handles a single source event. For all incoming events, it produces a chain that needs
 // to be executed and schedules the chain for execution.
-func (pe *planExecutor) handleSingleEvent(event SourceEvent) result.Result {
+func (pe *planExecutor) handleSingleEvent(updateTargetsOpt map[resource.URN]bool, event SourceEvent) result.Result {
 	contract.Require(event != nil, "event != nil")
 
 	var steps []Step
 	var res result.Result
 	switch e := event.(type) {
 	case RegisterResourceEvent:
 		logging.V(4).Infof("planExecutor.handleSingleEvent(...): received RegisterResourceEvent")
-		steps, res = pe.stepGen.GenerateSteps(e)
+		steps, res = pe.stepGen.GenerateSteps(updateTargetsOpt, e)
 	case ReadResourceEvent:
 		logging.V(4).Infof("planExecutor.handleSingleEvent(...): received ReadResourceEvent")
 		steps, res = pe.stepGen.GenerateReadSteps(e)
```

Comment:
this feels like something we need to consider.  if we're asking to read a resource that was not in the 'update' set, what should we do.  Would it seem reasonable to state that we don't do the read here?  should we generate a same-step for a read?  it's not quite clear to me what's appropriate for reads.